### PR TITLE
ocamlPackages.get-activity: init at 2.0.1

### PR DESCRIPTION
**Pull request by zazedd**
Changelog: https://github.com/tarides/get-activity/releases/tag/2.0.1

### Fixed
- Filter out null elements that are sometimes in the nodes list (https://github.com/tarides/get-activity/pull/39, @gpetiot)

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/ocaml-modules/get-activity/lib.nix [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  src = fetchFromGitHub {
+    owner = "tarides";
+    repo = "get-activity";
+    rev = version;
+    hash = "sha256-QU/LPIxcem5nFvSxcNApOuBu6UHqLHIXVSOJ2UT0eKA=";
+  };
+
+  minimalOCamlVersion = "4.08";
+
+  buildInputs = [ ppx_expect alcotest ];
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/get-activity/lib.nix:28]:
```suggestion
    checkInputs = [ ppx_expect alcotest ];
```

File: pkgs/development/ocaml-modules/get-activity/lib.nix [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+  propagatedBuildInputs = [
+    astring
+    curly
+    fmt
+    logs
+    ppx_yojson_conv
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/get-activity/lib.nix:35]: Propagating `ppx_yojson_conv_lib` (instead of `ppx_yojson_conv`) is enough.

Comment (zazedd) [reply pkgs/development/ocaml-modules/get-activity/lib.nix:35]: How did you figure this out?

Comment (vbgl) [reply pkgs/development/ocaml-modules/get-activity/lib.nix:35]: Reading the `META` file:

> cat /nix/store/…-ocaml5.1.1-get-activity-lib-2.0.1/lib/ocaml/5.1.1/site-lib/get-activity-lib/META
description = ""
requires = "astring curly fmt logs ppx_yojson_conv_lib yojson"

File: pkgs/development/ocaml-modules/get-activity/default.nix [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  alcotest
+}:
+
+buildDunePackage rec {
+  pname = "get-activity";
+  inherit (get-activity-lib) version src;
+
+  minimalOCamlVersion = "4.08";
+
+  propagatedBuildInputs = [
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/get-activity/default.nix:19]:
```suggestion
  buildInputs = [
```

Comment (zazedd) [reply pkgs/development/ocaml-modules/get-activity/default.nix:19]: There was no `META` file, so that got me confused. Thanks!

**Pull request merged**